test/jdk/javax/swing/JTableHeader/TableHeaderBorderPositionTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.imageio.ImageIO;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.JTableHeader;
+import javax.swing.UIManager;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+import java.io.File;
+import java.io.IOException;
+
+/*
+ * @test
+ * @bug 8311031
+ * @summary Test to validate JTable header border vertical line is
+ *          aligned with data grid lines (Metal L&F).
+ * @run main TableHeaderBorderPositionTest
+ */
+public class TableHeaderBorderPositionTest {
+    private static final double SCALE = 2.25;
+    private static boolean testStatus = true;
+    private static String failureMsg;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        SwingUtilities.invokeAndWait(TableHeaderBorderPositionTest::test);
+        if (!testStatus) {
+            throw new RuntimeException(failureMsg);
+        }
```

Comment:
```suggestion
        if (failureMsg != null) {
            throw new RuntimeException(failureMsg);
        }
```

`testStatus` is redudant.
`failureMsg` is to be declared `volatile` because it's used on different threads; however, `invokeAndWait` internally synchronises, so I think it's fine but I'm still unsure about it and I couldn't find any confirmation or disproof thereof.